tests/test_trainer.py

```diff
@@ -206,16 +206,21 @@ def get_regression_trainer(a=0, b=0, double_output=False, train_len=64, eval_len
         label_names = kwargs.get("label_names", None)
         train_dataset = RegressionDataset(length=train_len, label_names=label_names)
         eval_dataset = RegressionDataset(length=eval_len, label_names=label_names)
-        if pretrained:
-            config = RegressionModelConfig(a=a, b=b, double_output=double_output)
-            model = RegressionPreTrainedModel(config)
+
+        model_init = kwargs.pop("model_init", None)
+        if model_init is not None:
+            model = None
         else:
-            model = RegressionModel(a=a, b=b, double_output=double_output)
+            if pretrained:
+                config = RegressionModelConfig(a=a, b=b, double_output=double_output)
+                model = RegressionPreTrainedModel(config)
+            else:
+                model = RegressionModel(a=a, b=b, double_output=double_output)
+
```

Comment:
Note, that I found this buglet as I first tried to use `model_init` - `get_regression_trainer` was passing both `model` and `model_init` which means that `model_init` would never be run. At the end I didn't need `model_init` but I thought I'd leave the fix for future uses.